presto-thrift-interface/src/test/java/com/facebook/presto/thrift/interfaces/readwrite/TestColumnReaderWriter.java

```diff
@@ -0,0 +1,352 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.readwrite;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.BooleanType;
+import com.facebook.presto.spi.type.DoubleType;
+import com.facebook.presto.spi.type.IntegerType;
+import com.facebook.presto.spi.type.TimestampType;
+import com.facebook.presto.spi.type.TinyintType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.VarcharType;
+import com.facebook.presto.thrift.interfaces.client.ThriftColumnData;
+import com.facebook.presto.thrift.interfaces.client.ThriftRowsBatch;
+import com.facebook.presto.thrift.interfaces.readers.ColumnReaders;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriter;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriters;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.facebook.presto.spi.type.VarcharType.createUnboundedVarcharType;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestColumnReaderWriter
+{
+    private static final double NULL_FRACTION = 0.1;
+    private static final int MAX_VARCHAR_GENERATED_LENGTH = 32;
+    private static final char[] SYMBOLS;
+    private static final long MIN_GENERATED_TIMESTAMP;
+    private static final long MAX_GENERATED_TIMESTAMP;
+    private final AtomicLong seedGenerator = new AtomicLong(762103512L);
+
+    static {
+        char[] symbols = new char[2 * 26 + 10];
+        int next = 0;
+        for (char ch = 'A'; ch <= 'Z'; ch++) {
+            symbols[next++] = ch;
+        }
+        for (char ch = 'a'; ch <= 'z'; ch++) {
+            symbols[next++] = ch;
+        }
+        for (char ch = '0'; ch <= '9'; ch++) {
+            symbols[next++] = ch;
+        }
+        SYMBOLS = symbols;
+
+        Calendar calendar = Calendar.getInstance();
+
+        calendar.set(2000, Calendar.JANUARY, 1);
+        MIN_GENERATED_TIMESTAMP = calendar.getTimeInMillis();
+
+        calendar.set(2020, Calendar.DECEMBER, 31);
+        MAX_GENERATED_TIMESTAMP = calendar.getTimeInMillis();
+    }
+
+    @Test(invocationCount = 10)
```

Comment:
It's random with a fixed seed, so input is deterministic. Generated input is a good way to cover a lot of cases, so I'd like to keep it. There are manually created tests to cover some cases for LongReader and LongWriter.